File: migrations/populate.py
import sqlite3
import requests
import time
import subprocess
import os
import json

# Start the server
print("Starting backend server...")
server = subprocess.Popen(["go", "run", "backend/main.go"], cwd="C:\\Users\\aryan\\repos\\PCMaxing_Go")
time.sleep(3) # wait for server to start

try:
    users = []
    # Create 4 users
    for i in range(1, 5):
        email = f"test{i}@pcmaxing.xyz"
        username = f"test{i}"
        password = "password123"
        print(f"Creating user {username}...")
        resp = requests.post("http://localhost:8080/api/auth/signup", json={
            "email": email,
            "username": username,
            "password": password
        })
        if resp.status_code == 200:
            print(f"Success creating {username}")
        else:
            print(f"Failed or already exists {username}:", resp.text)
            # Try to login instead to get token if needed, but we don't need token for popular builds
        users.append(username)
    
    # Now we insert 10 builds into popular_builds
    db_path = "C:\\Users\\aryan\\repos\\PCMaxing_Go\\pcmaxing.db"
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    cursor.execute("DELETE FROM popular_builds") # clear existing
    
    # We need some parts, let's fetch one from API
    parts = []
    resp = requests.get("http://localhost:8080/api/components")
    if resp.status_code == 200:
        comps = resp.json()
        if comps:
            cpu = next((c for c in comps if c['category'] == 'CPU'), None)
            gpu = next((c for c in comps if c['category'] == 'GPU'), None)
            if cpu: parts.append(cpu)
            if gpu: parts.append(gpu)
    
    parts_json = json.dumps(parts) if parts else "[]"
    
    tiers = ["Budget", "Mid", "High", "Enthusiast"]
    
    for i in range(1, 11):
        user = users[i % 4]
        name = f"Build #{i} by {user}"
        tier = tiers[i % 4]
        price = 50000 + i * 10000
        
        cursor.execute('''
            INSERT INTO popular_builds (name, description, parts, total_price, tier)
            VALUES (?, ?, ?, ?, ?)
        ''', (name, f"A great {tier} PC build created by our user {user}.", parts_json, price, tier))
    
    conn.commit()
    conn.close()
    print("Successfully populated 10 popular builds.")
    
finally:
    server.terminate()

File: migrations/populate.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"math/rand"
	"net/http"
	"time"
)

const baseURL = "http://localhost:5000/api"

type User struct {
	Username string
	Email    string
	Password string
}

func main() {
	rand.Seed(time.Now().UnixNano())

	// Fetch categories and components
	resp, err := http.Get(baseURL + "/components")
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	var allComps []map[string]interface{}
	json.NewDecoder(resp.Body).Decode(&allComps)

	compsByCat := make(map[string][]map[string]interface{})
	for _, c := range allComps {
		cat := c["category"].(string)
		compsByCat[cat] = append(compsByCat[cat], c)
	}

	fmt.Printf("Loaded components for %d categories\n", len(compsByCat))

	// Create 5 users
	var tokens []string
	for i := 1; i <= 5; i++ {
		u := User{
			Username: fmt.Sprintf("user%d_%d", i, rand.Intn(10000)),
			Email:    fmt.Sprintf("user%d_%d@test.com", i, rand.Intn(10000)),
			Password: "password123",
		}
		
		body, _ := json.Marshal(map[string]string{
			"email":    u.Email,
			"username": u.Username,
			"password": u.Password,
		})
		
		res, err := http.Post(baseURL+"/auth/signup", "application/json", bytes.NewBuffer(body))
		if err != nil {
			fmt.Println("Error signing up:", err)
			continue
		}
		
		var authRes map[string]interface{}
		json.NewDecoder(res.Body).Decode(&authRes)
		res.Body.Close()
		
		if token, ok := authRes["token"].(string); ok {
			tokens = append(tokens, token)
			fmt.Println("Created user:", u.Username)
		} else {
			fmt.Println("Signup failed:", authRes)
		}
	}

	if len(tokens) == 0 {
		fmt.Println("No users created, exiting")
		return
	}

	// Create 50 builds
	categories := []string{"CPU", "GPU", "Motherboard", "RAM", "Storage", "PSU", "Case"}
	
	for i := 1; i <= 50; i++ {
		token := tokens[rand.Intn(len(tokens))]
		
		parts := make(map[string]interface{})
		totalPrice := 0.0
		
		for _, cat := range categories {
			catComps := compsByCat[cat]
			if len(catComps) > 0 {
				comp := catComps[rand.Intn(len(catComps))]
				parts[cat] = comp
				if price, ok := comp["price"].(float64); ok {
					totalPrice += price
				}
			}
		}

		// Check compatibility
		compatBody, _ := json.Marshal(map[string]interface{}{"parts": parts})
		compatRes, err := http.Post(baseURL+"/compatibility", "application/json", bytes.NewBuffer(compatBody))
		if err == nil {
			var compat map[string]interface{}
			json.NewDecoder(compatRes.Body).Decode(&compat)
			compatRes.Body.Close()
			if issues, ok := compat["issues"].([]interface{}); ok && len(issues) > 0 {
				fmt.Printf("Build %d has %d compatibility issues\n", i, len(issues))
			}
		}

		// Save build
		buildBody, _ := json.Marshal(map[string]interface{}{
			"name":        fmt.Sprintf("Awesome Build %d", i),
			"parts":       parts,
			"total_price": totalPrice,
			"is_public":   true,
		})
		
		req, _ := http.NewRequest("POST", baseURL+"/builds/saved", bytes.NewBuffer(buildBody))
		req.Header.Set("Content-Type", "application/json")
		req.Header.Set("Authorization", "Bearer "+token)
		
		client := &http.Client{}
		saveRes, err := client.Do(req)
		if err == nil {
			saveRes.Body.Close()
			if saveRes.StatusCode == 200 {
				fmt.Printf("Successfully created build %d\n", i)
			} else {
				b, _ := ioutil.ReadAll(saveRes.Body)
				fmt.Printf("Failed to create build %d: %s\n", i, string(b))
			}
		}
	}
	fmt.Println("Done creating 50 builds.")
}
